### PR TITLE
Fix completion checking

### PR DESCRIPTION
**Pull request by NigelCunningham**
In using imagesLoaded with Masonry for a client, the masonry resize was sometimes seen to fire before the images had loaded, causing the div to end up with too small a size. The issue is easily reproducible when using Chrome's bandwidth limiting and caching disabled in devtools.

This series of patch modifies imagesLoaded so it correctly handles partially loaded images, and only fires callbacks when all images really are loaded. To achieve this, I've used an interval timer. I'd prefer an event based solution but understand that image load events don't bubble. :(

### CODE REVIEW

File: imagesloaded.js [hunk 1]
```diff
@@ -198,37 +200,48 @@ ImagesLoaded.prototype.addBackground = function( url, elem ) {
...
+      }
     });
-  }
+  };
 
-  this.images.forEach( function( loadingImage ) {
-    loadingImage.once( 'progress', onProgress );
-    loadingImage.check();
-  });
+  this.intervalCheck = setInterval(this.runCheck, 500);
```
Comment (desandro) [inline imagesloaded.js]: Thanks for this contribution! I like some of the improvements, but this use of setInterval has me concerned. I'd rather not have a running interval in use at all.

Comment (NigelCunningham) [inline imagesloaded.js]: Sorry for the delay in getting back to you.
I agree - I'd prefer things to be event based wherever possible. I've forgotten the logic behind using a timer now; I'll see if I can find some time to look at it some more.

**Comment by NigelCunningham** [comment]
Hi again.

Here's an updated version that makes the whole thing event based. The only timers left are the original ones that have comments saying they work around Chrome issues.

Regards,

Nigel

**Comment by desandro** [comment]
Thanks for updating this! I'll keep this PR open to see if this is useful for others.

**Comment by NigelCunningham** [comment]
Thanks for the reply. Is there some reason why you think it shouldn't be merged? If so, I'll be happy to address that.

**Comment by desandro** [comment]
I'd like to see who else out there has found this useful.

**Comment by NigelCunningham** [comment]
Ah, ok :) Thanks.

**Pull request closed without merging**